server.ts

```diff
@@ -66,9 +73,33 @@ if (args.startSimple) {
     createSimpleServer(args);
 }
 
+function getBuildsLinks(userAgent: string = ''): Array<{ url: string; text: string }> {
+    const result = [];
+    const platform = userAgent.includes('Electron') ? 'desktop' : 'web';
+
+    connectionTypes.forEach((connection) => {
+        buildTypes.forEach((build) => {
+            result.push({
+                url: `/choose/${platform}/${connection}/${build}`,
+                text: `${platform} ${connection} ${build}`
+            });
+        });
+    });
 
-function arrToHash(arr: Array<string>): Object {
-    const result = Object.create(null);
-    arr.forEach((some) => result[some] = true);
     return result;
 }
+
+function parseCookie(header = ''): IRequestData {
+    const [platform, connection, build] = ((parserCookie(header) || Object.create(null)).session || '').split(',');
+    if (!build || !connection || !platform) {
```

Comment:
Applying De Morgan's law to switch `!build || !connection || !platform` to `!(build && connection && platform)` might make it clearer that all three values are required. Or might not:)